dreambooth/utils/image_utils.py

```diff
@@ -0,0 +1,431 @@
+
+        if os.path.exists(text_filename):
+            with open(text_filename, "r", encoding="utf8") as file:
+                filename_text = file.read().strip()
+        else:
+            filename_text = os.path.splitext(filename)[0]
+            filename_text = re.sub(self.re_numbers_at_start, '', filename_text)
+            if self.re_word:
+                tokens = self.re_word.findall(filename_text)
+                filename_text = (shared.dataset_filename_join_string or "").join(tokens)
+
+        filename_text = re.sub(r'\\', "", filename_text)  # work with \(franchies\)
+        return filename_text
+
+    def create_text(self, text_template, filename_text, instance_token, class_token, is_class=True):
+        # Append the filename text to the template first...THEN shuffle it all.
+        filename_text = text_template.replace("[filewords]", filename_text)
+        # If we are creating text for a class image and it has our instance token in it, remove/replace it
+        class_tokens = [f"a {class_token}", f"the {class_token}", f"an {class_token}", class_token]
+        if instance_token != "" and class_token != "":
+            reg_inst = f"\\b{instance_token}\\b"
+            reg_class = f"\\b{class_token}\\b"
+
+            if is_class and re.search(reg_inst, filename_text):
+                if re.search(reg_class, filename_text):
+                    filename_text = re.sub(reg_inst, "", filename_text)
+                else:
+                    filename_text = re.sub(reg_inst, class_token, filename_text)
+
+            if not is_class:
+                if re.search(reg_class, filename_text):
+                    # Do nothing if we already have class and instance in string
+                    if re.search(reg_inst, filename_text):
+                        pass
+                    # Otherwise, substitute class tokens for the base token
+                    else:
+                        for token in class_tokens:
+                            reg_token = f"\\b{token}\\b"
+                            filename_text = re.sub(reg_token, class_token, filename_text)
+
+                    # Now, replace class with instance + class tokens
+                    filename_text = re.sub(reg_class, f"{instance_token} {class_token}", filename_text)
+                else:
+                    # If class is not in the string, check if instance is
+                    if re.search(reg_inst, filename_text):
+                        filename_text = re.sub(reg_inst, f"{instance_token} {class_token}", filename_text)
+                    else:
+                        # Description only, insert both at the front?
+                        filename_text = f"{instance_token} {class_token}, {filename_text}"
+
+        # Remove underscores, double-spaces, and other characters that will cause issues.
+        filename_text = re.sub(r"_| +", " ", filename_text)
+        filename_text = re.sub(r"[^\w ]", "", filename_text)
```

Comment:
Need to preserve comma as below.
filename_text = re.sub(r"[^\w ,]", "", filename_text)